core/src/main/java/com/linecorp/armeria/common/HttpResponse.java

```diff
@@ -171,7 +173,12 @@ static HttpResponse from(CompletionStage<? extends HttpResponse> stage) {
         final DeferredHttpResponse res = new DeferredHttpResponse();
         stage.whenComplete((delegate, cause) -> {
             if (cause != null) {
-                res.close(cause);
+                Throwable rootCause = Throwables.getRootCause(cause);
```

Comment:
CompletableFuture should wrap exception with `CompletionException` . So make HttpResponse + handler can handle HttpResponseException.